k8s-operator/sessionrecording/ws/conn.go

```diff
@@ -140,17 +170,32 @@ func (c *conn) Read(b []byte) (int, error) {
 	}
 	c.readBuf.Next(len(readMsg.raw))
 
-	if readMsg.isFinalized {
+	if readMsg.isFinalized && !c.readMsgIsIncomplete() {
 		// Stream IDs for websocket streams are static.
 		// https://github.com/kubernetes/client-go/blob/v0.30.0-rc.1/tools/remotecommand/websocket.go#L218
 		if readMsg.streamID.Load() == remotecommand.StreamResize {
-			var err error
 			var msg tsrecorder.ResizeMsg
 			if err = json.Unmarshal(readMsg.payload, &msg); err != nil {
 				return 0, fmt.Errorf("error umarshalling resize message: %w", err)
 			}
+
 			c.ch.Width = msg.Width
 			c.ch.Height = msg.Height
+
+			// If this is initial resize message, the width and
+			// height will be sent in the CastHeader. If this is a
+			// subsequent resize message, we need to send asciinema
+			// resize message.
+			var isInitialResize bool
+			c.sendInitialTermSizeSetOnce.Do(func() {
+				isInitialResize = true
+				c.initialTermSizeSet <- "set" // unblock sending of CastHeader
```

Comment:
> I pushed an untested example in https://github.com/tailscale/tailscale/commit/faff089f132cf22d01167ce5a87282fefea27519 to try and illustrate.

We need to write cast header for non-interactive sessions that don't get get resize messages, so this will not work (I initially wrote similar code for this)

Reply:
Thanks, that's better, I missed these suggestions in the previous review